src/concurrency.md

```diff
@@ -232,6 +233,96 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[threadpool-walk]: #threadpool-walk
+<a name="threadpool-walk"></a>
+
+## Calculate SHA1 sum of *.iso files concurrently
+
+[![threadpool-badge]][threadpool] [![num_cpus-badge]][num_cpus] [![walkdir-badge]][walkdir] [![cat-concurrency-badge]][cat-concurrency][![cat-filesystem-badge]][cat-filesystem]
+
+This example uses `threadpool`, `walkdir` and `ring` crates to calculate the SHA1 for every file present in the current directory. A threadpool `pool` is created using the number of cpus present in the system. Then every `entry` returned by `walkdir` is passed into this pool to perform the operations of reading and compute SHA1. At the end the program waits for all jobs to finish. To get better results, compile this program in release mode.
```

Comment:
1. It would replace

> "This example uses `threadpool`, `walkdir` and `ring` crates to calculate ..."

with "Calculates ...". The reader already knows which crates are used form the heading.

2. Actually this example recursively calculates SHA256 of every iso file present in "/home/user/Downloads"

So please update the code or the description ;)

3. typo - "reading and compute" => "reading and computing" 

4. There are several identifiers mentioned `pool`, `entry` etc. Please link these to docs